src/util/createInstantSearch.js

```diff
@@ -75,19 +143,40 @@ export const createInstantSearch = instantSearchOptions => {
       warn('The result of `getState()` needs to be passed to `hydrate()`.');
       return;
     }
-    const { lastResults } = instantSearchState;
-    search.searchParameters = lastResults._state;
+    const { lastResults, derived } = instantSearchState;
     search.helper = algoliaHelper(searchClient, indexName, lastResults._state);
     search.helper.lastResults = new SearchResults(
       new SearchParameters(lastResults._state),
       lastResults._rawResults
     );
 
+    // TODO: this is still causing _two_ close to identical requests on the frontend
+    // maybe a mismatch of parameters.
+    derivedHelpers = Object.keys(derived).reduce((acc, indexId) => {
+      if (indexId === indexName) {
+        return acc;
+      }
+
+      // eslint-disable-next-line no-param-reassign
+      acc[indexId] = {
+        // __forceRender uses lastResults only, we don't want to duplicate
+        // the requests by doing a full derive here.
+        lastResults: new SearchResults(
```

Comment:
instead of passing it this way and using index.hydrate later, InstantSearch could have an API which accepts indexId / SearchResults map. We can use that both for hydrate & for findResultsState